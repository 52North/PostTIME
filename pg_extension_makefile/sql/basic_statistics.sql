
SELECT pt_histogram_intervals('2012/2020','P1Y');

SELECT pt_histogram_intervals('2012-3-5/2020-8-5','P1Y');

SELECT pt_histogram_intervals('2012-3-5/2020-8-5','P1Y6M');

SELECT pt_histogram_intervals('TCS0021/1000000000','P1Y6M');

SELECT pt_histogram_intervals('2012-3-5','P1Y6M');

SELECT pt_histogram_intervals('2012-3-5,2020-8-5','P1Y6M');

SELECT pt_histogram_intervals('ORD001Triassic','P1Y6M');