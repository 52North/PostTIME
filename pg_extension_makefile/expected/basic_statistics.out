SELECT pt_histogram_intervals('2012/2020','P1Y');
                 pt_histogram_intervals                  
---------------------------------------------------------
 CAL0012012-01-01T00:00:00.000Z/2013-01-01T00:00:00.000Z
 CAL0012013-01-01T00:00:00.000Z/2014-01-01T00:00:00.000Z
 CAL0012014-01-01T00:00:00.000Z/2015-01-01T00:00:00.000Z
 CAL0012015-01-01T00:00:00.000Z/2016-01-01T00:00:00.000Z
 CAL0012016-01-01T00:00:00.000Z/2017-01-01T00:00:00.000Z
 CAL0012017-01-01T00:00:00.000Z/2018-01-01T00:00:00.000Z
 CAL0012018-01-01T00:00:00.000Z/2019-01-01T00:00:00.000Z
 CAL0012019-01-01T00:00:00.000Z/2020-01-01T00:00:00.000Z
 CAL0012020-01-01T00:00:00.000Z/2021-01-01T00:00:00.000Z
(9 rows)

SELECT pt_histogram_intervals('2012-3-5/2020-8-5','P1Y');
                 pt_histogram_intervals                  
---------------------------------------------------------
 CAL0012012-01-01T00:00:00.000Z/2013-01-01T00:00:00.000Z
 CAL0012013-01-01T00:00:00.000Z/2014-01-01T00:00:00.000Z
 CAL0012014-01-01T00:00:00.000Z/2015-01-01T00:00:00.000Z
 CAL0012015-01-01T00:00:00.000Z/2016-01-01T00:00:00.000Z
 CAL0012016-01-01T00:00:00.000Z/2017-01-01T00:00:00.000Z
 CAL0012017-01-01T00:00:00.000Z/2018-01-01T00:00:00.000Z
 CAL0012018-01-01T00:00:00.000Z/2019-01-01T00:00:00.000Z
 CAL0012019-01-01T00:00:00.000Z/2020-01-01T00:00:00.000Z
 CAL0012020-01-01T00:00:00.000Z/2021-01-01T00:00:00.000Z
(9 rows)

SELECT pt_histogram_intervals('2012-3-5/2020-8-5','P1Y6M');
                 pt_histogram_intervals                  
---------------------------------------------------------
 CAL0012012-03-01T00:00:00.000Z/2013-09-01T00:00:00.000Z
 CAL0012013-09-01T00:00:00.000Z/2015-03-01T00:00:00.000Z
 CAL0012015-03-01T00:00:00.000Z/2016-09-01T00:00:00.000Z
 CAL0012016-09-01T00:00:00.000Z/2018-03-01T00:00:00.000Z
 CAL0012018-03-01T00:00:00.000Z/2019-09-01T00:00:00.000Z
 CAL0012019-09-01T00:00:00.000Z/2021-03-01T00:00:00.000Z
(6 rows)

SELECT pt_histogram_intervals('TCS0021/1000000000','P1Y6M');
                 pt_histogram_intervals                  
---------------------------------------------------------
 CAL0021969-12-01T00:00:00.000Z/1971-06-01T00:00:00.000Z
 CAL0021971-06-01T00:00:00.000Z/1972-12-01T00:00:00.000Z
 CAL0021972-12-01T00:00:00.000Z/1974-06-01T00:00:00.000Z
 CAL0021974-06-01T00:00:00.000Z/1975-12-01T00:00:00.000Z
 CAL0021975-12-01T00:00:00.000Z/1977-06-01T00:00:00.000Z
 CAL0021977-06-01T00:00:00.000Z/1978-12-01T00:00:00.000Z
 CAL0021978-12-01T00:00:00.000Z/1980-06-01T00:00:00.000Z
 CAL0021980-06-01T00:00:00.000Z/1981-12-01T00:00:00.000Z
 CAL0021981-12-01T00:00:00.000Z/1983-06-01T00:00:00.000Z
 CAL0021983-06-01T00:00:00.000Z/1984-12-01T00:00:00.000Z
 CAL0021984-12-01T00:00:00.000Z/1986-06-01T00:00:00.000Z
 CAL0021986-06-01T00:00:00.000Z/1987-12-01T00:00:00.000Z
 CAL0021987-12-01T00:00:00.000Z/1989-06-01T00:00:00.000Z
 CAL0021989-06-01T00:00:00.000Z/1990-12-01T00:00:00.000Z
 CAL0021990-12-01T00:00:00.000Z/1992-06-01T00:00:00.000Z
 CAL0021992-06-01T00:00:00.000Z/1993-12-01T00:00:00.000Z
 CAL0021993-12-01T00:00:00.000Z/1995-06-01T00:00:00.000Z
 CAL0021995-06-01T00:00:00.000Z/1996-12-01T00:00:00.000Z
 CAL0021996-12-01T00:00:00.000Z/1998-06-01T00:00:00.000Z
 CAL0021998-06-01T00:00:00.000Z/1999-12-01T00:00:00.000Z
 CAL0021999-12-01T00:00:00.000Z/2001-06-01T00:00:00.000Z
 CAL0022001-06-01T00:00:00.000Z/2002-12-01T00:00:00.000Z
(22 rows)

SELECT pt_histogram_intervals('2012-3-5','P1Y6M');
                 pt_histogram_intervals                  
---------------------------------------------------------
 CAL0012012-03-01T00:00:00.000Z/2013-09-01T00:00:00.000Z
(1 row)

SELECT pt_histogram_intervals('2012-3-5,2020-8-5','P1Y6M');
ERROR:  This functions works only with periods.
SELECT pt_histogram_intervals('ORD001Triassic','P1Y6M');
ERROR:  This function does not work with Ordinal Systems
